### DISCUSSION TITLE
Group 13

### DISCUSSION THREAD

**Discussion by nicolehartland** [Digital-Preservation-Coalition/PythonStudyGroups#10] [General]
-

**Comment by barsness13** [comment]
here's our google drive folder:

https://drive.google.com/drive/folders/1hdcXvolg5XXi3Ttl08on9gURZRHCfx9i?usp=drive_link

**Comment by digitensions** [comment]
Hi group 13! Hope things are going well with your study groups! I'm about this month if I can help at all, just ask. Joanna

**Comment by NappierMarcus** [comment]
I've added a test python script for the file/folder rename to the Google Drive. Happy to work through that on Wednesday!

**Reply by barsness13** [reply]
Awesome!  I added the script I made for parsing RSS data as well, in case anybody wants another script to look at.

**Comment by nicolehartland** [comment]
Hi everyone - I unfortunately can't make the next session but thanks for adding scripts etc. - been really useful to look at alongside refamiliarising myself with the basics and I feel a bit more comfortable with Python now. Will catch up on what you discuss this month !

**Comment by mpstrickland** [comment]
Hi everyone, I have added a couple of desiderata in our spreadsheet. I was wondering whether any of you has ever used python to convert *.pst to *.mbox. We are working with our Records Management team to profile a workflow for email transfer and I am sure that there are ways in which I could use python to automate it. Thank you, and see you next week! :)

**Reply by digitensions** [reply]
Looking on Pypi.org there's a library that may help:
https://pypi.org/project/mailbagit/

**Reply by mpstrickland** [reply]
I forgot to thank you... I just realised, sorry! 😓

**Comment by barsness13** [comment]
One of my bits of homework was to mess around with Marcus' renaming script -- we wanted to try and add some error handling and reporting.  I just uploaded my version to the Google Drive: https://drive.google.com/file/d/1ewOZY5Ag9Teo5S-McfrMQdWhQhPSDc14/view?usp=drive_link  I wanted to share it in advance in case anybody wanted to poke around in the code, but I'm also happy to walk through it on the 27th if that's of any interest.

**Reply by barsness13** [reply]
woops, wrong file!  here's the correct one: https://drive.google.com/file/d/1LhwSQnook-RbnVxMpOcNtCi_5vDeWERF/view?usp=drive_link

**Reply by barsness13** [reply]
I realized that you can actually use it as written to move files as well, so I got to futzing.  I added the ability to create new file paths to reorganize files if desired.  I also updated the error handling a little bit, so that it checks the CSV for header rows and ensures that all the source files are in place before the renaming/moving begins.  https://drive.google.com/file/d/14GfnXfh4ukhAyeTZCOZ0joLYiiWiJvLF/view?usp=drive_link

**Comment by barsness13** [comment]
last session some of you expressed interest in a script a colleague of mine was writing -- it compares two CSV files with hash values and filenames, and returns info on all the files that are in the "old" file list but not in the "new."  It's still very much a draft, but there was enough interest that she agreed to let me share it with you all: https://drive.google.com/file/d/1Qeu_6L3ZzdyRbFhtDZ3i_-FKVszrQg9Q/view?usp=drive_link

**Comment by NappierMarcus** [comment]
Anyone having issues getting into the zoom?

**Comment by barsness13** [comment]
Yep.  My coworker is also stuck in the waiting room, as is someone else
from her group.

*Sarah Barsness* (she, her, hers)
Digital Records Archivist
University Archives <https://www.lib.umn.edu/collections/special/uarchives>
 & University Digital Conservancy <https://conservancy.umn.edu/>
Elmer L Andersen Library <https://www.lib.umn.edu/collections/special> @
University of Minnesota

*Please let me know how I can make our time together more accessible*

**Reply by NappierMarcus** [reply]
gah! I have to run to a meeting at 10 so hopefully this gets sorted soon

**Comment by mpstrickland** [comment]
I have it on my calendar at 2pm (here local time, i.e., in 1 hour).

**Reply by NappierMarcus** [reply]
oh weird! I thought these started at 9am EST (my local time)

**Comment by barsness13** [comment]
I wonder if daylight savings time has gotten everything messed up?

*Sarah Barsness* (she, her, hers)
Digital Records Archivist
University Archives <https://www.lib.umn.edu/collections/special/uarchives>
 & University Digital Conservancy <https://conservancy.umn.edu/>
Elmer L Andersen Library <https://www.lib.umn.edu/collections/special> @
University of Minnesota

*Please let me know how I can make our time together more accessible*

**Reply by NappierMarcus** [reply]
oh yea...how did I forget about that. well I will miss roughly the first 30 mins of our session but should be able to hop in at 10:30 or so.

**Comment by barsness13** [comment]
The parts of the US that have daylight savings already had it, but the UK
doesn't have it until March 31.

*Sarah Barsness* (she, her, hers)
Digital Records Archivist
University Archives <https://www.lib.umn.edu/collections/special/uarchives>
 & University Digital Conservancy <https://conservancy.umn.edu/>
Elmer L Andersen Library <https://www.lib.umn.edu/collections/special> @
University of Minnesota

*Please let me know how I can make our time together more accessible*

**Comment by fullerpault** [comment]
I spent some time this morning working on the csvrename script and uploaded my version  to the shared folder. Take a look and let me know what you think! [csvrename_paul.py](https://drive.google.com/file/d/1IuZ8CBXkP35xd_xoKJEtyzrdXiMegHCY/view?usp=sharing)

**Comment by mpstrickland** [comment]
Hi all,
I am at a conference for the next three days and will not be able to attend today’s session, so sorry about it ☹
I need to do more trying with Paul’s script because I couldn’t get it to work. When I am back to the office I will try again and this time I am going to make notes so I can share them with you, to understand what I am doing wrong.
Have fun during the session!
Manuela

Dr Manuela Pallotto Strickland | Metadata and Digital Preservation Coordinator | Archives & Research Collections | Libraries & Collections

King's College London | Strand | London WC2R 2LS | ***@***.******@***.***>
Tel: Please call me using MS Teams or Skype for Business, or email to arrange a call

W: https://www.kcl.ac.uk/library/collections/archives
T: twitter.com/KingsArchives<http://twitter.com/KingsArchives> and twitter.com/kingslibraries<http://twitter.com/kingslibraries>
Blog: blogs.kcl.ac.uk/kingscollections<http://blogs.kcl.ac.uk/kingscollections>

**Reply by fullerpault** [reply]
Oh gosh, I think the reason it didn't work for you is that I didn't write it correctly! Not sure if I uploaded an older version by mistake, or if I was just doing something weird when testing it, but there was definitely something wrong with it. I couldn't get it to work on my work computer, either!

I've made a few tweaks, though, and got it to work. That said, it isn't very user friendly, and I'd love to pick everyone's brain today to figure out best to get the input file and handle the working directories.

**Comment by barsness13** [comment]
Unfortunately I have to miss today's session, as well.  I look forward to seeing what you all worked on in the notes -- I should be back next time!

**Comment by NappierMarcus** [comment]
Hey all,

I may have to hop off the group check-in at 10:30 for a meeting but should be able to rejoin again at 11 if we’re still on. Unfortunately I haven’t been able to work on too much since our last session since work has been crazy and took some time off.

Marcus

**Comment by barsness13** [comment]
According to my calendar, we're meeting tomorrow!  I've been trying to turn our renaming script into something that iterates through a spreadsheet of filepaths to check that the file path and file name are both valid and free of any special characters.  I haven't gotten it to work yet -- I keep getting a traceback error, so I haven't been able to make sure my regular expressions are working as intended.  I'm going to keep plugging away at it, but if you have any great ideas or want to workshop it tomorrow, here's what I've got so far: https://drive.google.com/file/d/1ARawv4-38McdyZ2h3ARPnRnrZltp8RF5/view?usp=drive_link

**Reply by barsness13** [reply]
Whoops, had my calendar wrong.  BUT now I have more time to fix my code!  I've got the script itself working, but my regular expressions aren't behaving as I would expect.  In case you want to play around with the updated version, it's here: https://drive.google.com/file/d/1SJqfNovVhbc9ha5EB_8wFrX3pSrRZOcs/view?usp=drive_link

**Comment by NappierMarcus** [comment]
Hey guys! I too thought our check-in was last week and then accepted a bunch of meetings for tomorrow so my schedule is a little wonky and not sure I'll be able to join. :/

**Comment by mpstrickland** [comment]
I am a bit confused now: do we have a meeting tomorrow? I had the meeting on the calendar for last week, but nothing on tomorrow. Sarah, can you tell me at what time is the meeting on your calendar, please? Thank you 😊

Dr Manuela Pallotto Strickland | Metadata and Digital Preservation Coordinator | Archives & Research Collections | Libraries & Collections

King's College London | Strand | London WC2R 2LS | ***@***.******@***.***>
Tel: Please call me using MS Teams or Skype for Business, or email to arrange a call

W: https://www.kcl.ac.uk/library/collections/archives
T: twitter.com/KingsArchives<http://twitter.com/KingsArchives> and twitter.com/kingslibraries<http://twitter.com/kingslibraries>
Blog: blogs.kcl.ac.uk/kingscollections<http://blogs.kcl.ac.uk/kingscollections>

**Comment by fullerpault** [comment]
Hi folks,

I had this on my calendar for last week, too. I think if folks want to meet
up sometime this week you should! I'm unfortunately not able to make it
this month (who knew paternity leave wasn't just a vacation!), but I hope
to be able to make next month's meeting. Happy pythoning!

Cheers,

*Paul Fuller *(he/him)
Digital Projects Associate

Digital Projects | Libraries & Learning
digitalprojects.usm.maine.edu
(207) 780-4334
Glickman Library 521
University of Southern Maine

**Comment by barsness13** [comment]
I'm glad it wasn't just me that was confused -- I was logged in last Wednesday and waiting for the Zoom to start, and it just never did.  Right after I gave up, I got an email with the subject line "Python Study Groups: Would you like to continue?" that indicated the next meeting was tomorrow ([Wednesday, 29 May at 14:00 UTC](https://www.timeanddate.com/worldclock/fixedtime.html?msg=Python+Study+Group+-+2+-+May&iso=20240529T14&p1=1440)).

**Comment by mpstrickland** [comment]
Uhm, okay so I can attend tomorrow's meeting but from 14.30 onwards (30 minutes after the meeting has started), as I have other meetings happening. Are any of you guys going to be online?


Dr Manuela Pallotto Strickland | Metadata and Digital Preservation Coordinator | Archives & Research Collections | Libraries & Collections

King's College London | Strand | London WC2R 2LS | ***@***.******@***.***>
Tel: Please call me using MS Teams or Skype for Business, or email to arrange a call

W: https://www.kcl.ac.uk/library/collections/archives
T: twitter.com/KingsArchives<http://twitter.com/KingsArchives> and twitter.com/kingslibraries<http://twitter.com/kingslibraries>
Blog: blogs.kcl.ac.uk/kingscollections<http://blogs.kcl.ac.uk/kingscollections>

**Comment by barsness13** [comment]
I'll be there, but if nobody else can make it NBD!  I am still putzing about with my file name and path validator (trying to plug it into our rename-and-move script), so I can always use that time to keep putzing around.

**Comment by mpstrickland** [comment]
Hi all in group 13,
I have a zoom meeting pinned down for us today, possibly the last one for our group, but I am wondering whether the date is not right as no one is there to let me in. I did not create it, it automatically created itself and populated my calendar as part of the recurrent event/series. If you read this and you know today there is no meeting, can you please let me know, or I will stay with headphones for the next hour… Thanks so much!
Manuela

Dr Manuela Pallotto Strickland | Metadata and Digital Preservation Coordinator | Archives & Research Collections | Libraries & Collections

King's College London | Strand | London WC2R 2LS | ***@***.******@***.***>
Tel: Please call me using MS Teams or Skype for Business, or email to arrange a call

W: https://www.kcl.ac.uk/library/collections/archives
T: twitter.com/KingsArchives<http://twitter.com/KingsArchives> and twitter.com/kingslibraries<http://twitter.com/kingslibraries>
Blog: blogs.kcl.ac.uk/kingscollections<http://blogs.kcl.ac.uk/kingscollections>

**Reply by barsness13** [reply]
I have it on my calendar as being today, but starting in 1 hour.

**Reply by NappierMarcus** [reply]
Hey all! I also have the Zoom meeting down for 9am but wondering if it's starting at 10am this morning? I think there was some confusion in previous weeks about the start time because of daylight savings time?

Also I'll be roughly 30 mins late but should be able to join for a bit at 10:30.

**Comment by mpstrickland** [comment]
Oh my, thanks Sarah!!!!! It must be some kind of time difference, I am not sure.
Thank you so much, I will come out of it and will join in zoom again in 50 minutes 😊
I also saw Marcus’ email, so maybe there has been some confusion with timezone etc
Looking forward to seeing you in a bit!

Dr Manuela Pallotto Strickland | Metadata and Digital Preservation Coordinator | Archives & Research Collections | Libraries & Collections

King's College London | Strand | London WC2R 2LS | ***@***.******@***.***>
Tel: Please call me using MS Teams or Skype for Business, or email to arrange a call

W: https://www.kcl.ac.uk/library/collections/archives
T: twitter.com/KingsArchives<http://twitter.com/KingsArchives> and twitter.com/kingslibraries<http://twitter.com/kingslibraries>
Blog: blogs.kcl.ac.uk/kingscollections<http://blogs.kcl.ac.uk/kingscollections>

**Comment by mpstrickland** [comment]
[like]  Manuela Pallotto Strickland reacted to your message:
________________________________
